tests/strictness-checks/Partial.hs

```diff
@@ -0,0 +1,170 @@
+module Partial where
+
+import Control.DeepSeq
+import Control.Exception
+import Control.Monad.Trans.Class
+import Control.Monad.Trans.Reader
+import Control.Monad.Trans.State.Strict
+import Data.Bifunctor
+import Data.Bitraversable
+import qualified Data.List as List
+import Numeric.Natural
+import Test.QuickCheck (Arbitrary (arbitrary, shrink), Gen, frequency)
+
+import Prelude'
+
+-- | A specification for a value that might not be defined.
+newtype Specification value = Specification {partial :: Maybe value} deriving (Show, Functor, Foldable, Traversable)
```

Comment:
Could you possibly slap `{-# LANGUAGE DeriveTraversable #-}` and have `data Specification value = Undefined | Defined value deriving (Show, Functor, Foldable, Traversable)` without pattern synonyms?

Reply:
I am using the `newtype` on purpose here — it is a proof that the type in question is isomorphous to `Maybe`. _«Entities should not be multiplied without necessity.»_ Patterns are then there for their descriptive names.

Why should we not like pattern synonyms? Are there some compatibility issues?

Reply:
Mostly for the sake of not multiplying entities ;) I'd say that having two ways to say the same (one with a constructor and one with a pattern synonym) is a violation of Occam's razor. I don't think having an immediate proof of isomorphism is worth it, you can potentially restore it using generics (see https://hackage.haskell.org/package/generic-maybe-0.3.0.4/docs/Data-Generics-Maybe.html#t:MaybeLike for example).

I do not feel too strongly about it.